src/Repository.js

```diff
@@ -43,7 +44,34 @@ export default class Repository {
     return this.lernaJson && this.lernaJson.bootstrapConfig || {};
   }
 
+  get packages() {
+    if (!this._packages) {
+      this.buildPackageGraph();
+    }
+    return this._packages;
+  }
+
+  get filteredPackages() {
+    if (!this._filteredPackages) {
+      this.buildPackageGraph();
+    }
+    return this._filteredPackages;
+  }
+
+  get packageGraph() {
+    if (!this._packageGraph) {
+      this.buildPackageGraph();
```

Comment:
So the big argument against this structure is that these are getters, but depending on the order that you invoke `repository.packageGraph` and `repository.buildPackageGraph(...)`, you can end up with permanently different values for `packageGraph` (or one of the other fields controlled by this). Spooky! 

There's also a hand-wavier argument about who should be _responsible_ for filtering; I figured that the "Repository" class should probably be a predictable, consistent, immutable representation of the state of the repo, and filtering should be applied elsewhere, namely, in the commands themselves. If you just pulled out the filtering behavior of `buildPackageGraph` to the root `Command` you should still get it across commands, yeah?

Lastly, you can't pass arguments to getters!